inf-clojure.el

```diff
@@ -766,7 +768,7 @@ process buffer for a list of commands.)"
                            (inf-clojure--prompt-repl-type))))
         (message "Starting Clojure REPL via `%s'..." cmd)
         (with-current-buffer (apply #'make-comint
-                                    "inf-clojure" (car cmdlist) nil (cdr cmdlist))
+                                    process-buffer-name (car cmdlist) nil (cdr cmdlist))
```

Comment:
Are you sure that's needed? At first I thought so myself, but you might have noticed that I just reverted it, before you added it. :D I don't think that the process name has to be unique or derived from the buffer name. After all - it was always the same until now and this worked just fine.

Reply:
without this change inf-clojure would make a fundamental mode buffer I end up in a fundamental buffer named `*inf-clojure build-your-own-logic-engine*` and the correct inf-clojure repl buffer is hidden and named `**inf-clojure build-your-own-logic-engine**`
<img width="863" alt="image" src="https://user-images.githubusercontent.com/6377293/112004087-7a622000-8aef-11eb-8182-9dbb143832ad.png">

Reply:
prior to this change the string "inf-clojure" was passed in, and the buffer name we used was "*inf-clojure*" for this reason

Reply:
e.g. that's from IELM:

![image](https://user-images.githubusercontent.com/103882/112004266-4ce69800-8b2a-11eb-9f1b-90c4a6e9742c.png)

Seems to me there's no need for the names to match, but I might be missing something. I no longer remember comint's APIs well. :D

Reply:
Actually, you're right. That's the only place where we pass the buffer name anyways. :D I wonder what I was thinking when I deleted this. :D

Reply:
i suspect that's the disambiguation that they do when they do match. ie, the same name goes in and it adds a 1 on the end. I suspect the best thing to do is actually start up the repl and then rename it `repl-projec-name`

Reply:
You're probably right, I didn't dwell on this much. I was also thinking of making the naming template configurable in case someone wants different names by default.